### ISSUE TITLE
[stable/redis-ha] Cannot helm upgrade due to versionned matchLabels

### ISSUE THREAD

**Issue by roychri** [helm/charts#4513]
BUG REPORT

**Version of Helm and Kubernetes**:
Kubernetes: 1.8.4
Helm: 2.7.2

**Which chart**:
Chart: redis-ha

**What happened**:
Trying to upgrade a release produces an error because its trying to match based on a label that contains the version of the chart and when the chart has a new version, it cannot upgrade it since it cannot find the deploy/pods..
This is related to: https://github.com/kubernetes/helm/issues/2494

The error I got was:
```
$ helm upgrade redis-ha-dev stable/redis-ha --set rbac.create=false
Error: UPGRADE FAILED: Deployment.apps "redis-ha-dev-redis-ha-sentinel" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"name":"redis-ha-dev-redis-ha-sentinel", "release":"redis-ha-dev", "app":"redis-ha", "chart":"redis-ha-2.1.1", "heritage":"Tiller"}: `selector` does not match template `labels` && Deployment.apps "redis-ha-dev-redis-ha-server" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"release":"redis-ha-dev", "app":"redis-ha", "chart":"redis-ha-2.1.1", "heritage":"Tiller", "name":"redis-server", "redis-node":"true"}: `selector` does not match template `labels`
```

**What you expected to happen**:
I expected the chart to be upgraded without any errors.

**How to reproduce it** (as minimally and precisely as possible):

In short: Install a previous version and try to upgrade to the latest version:

$ helm install stable/redis-ha --name redis-ha-dev --version 2.0.0
$ helm upgrade redis-ha-dev stable/redis-ha --set rbac.create=false

**Anything else we need to know**:

This is because the chart name and version are present in the `.spec.template.metadata.labels` but should not be (see https://github.com/kubernetes/helm/issues/2494 for details).
Or, an alternative would be to have specific `.spec.selector.matchLabels` values that does not include the chart version.  Do you have any preference? I can make a PR once I know which one of those two is preferable...
